build.rs

```diff
@@ -17,6 +17,8 @@ fn main() {
 
     if !compiler.nightly {
         println!("cargo:rustc-cfg=syn_disable_nightly_tests");
+    } else {
+        println!("cargo:rustc-cfg=syn_enable_doc_cfg");
```

Comment:
Note that the compiler minor version doesn't need to be checked, since the `#[feature(doc_cfg)]` was introduced before rustc 1.21, which is way older than the MSRV of syn.